projects/netgrif-components-core/src/lib/task-content/services/field-converter.service.ts

```diff
@@ -36,30 +36,14 @@ export class FieldConverterService {
                 return new BooleanField(item.stringId, item.name, item.value as boolean, item.behavior,
                     item.placeholder, item.description, item.layout, item.validations, item.component, item.parentTaskId);
             case FieldTypeResource.TEXT:
-                /*@deprecated in 4.3.0*/
-                let type = TextFieldView.DEFAULT;
-                if (item.subType !== undefined && item.subType === 'area') {
-                    type = TextFieldView.TEXTAREA;
-                }
-                if (item.view !== undefined && item.view.value !== undefined && (item.view.value === 'editor' || item.view.value === TextFieldView.RICHTEXTAREA)) {
-                    type = TextFieldView.RICHTEXTAREA;
-                    return new TextAreaField(item.stringId, item.name, item.value as string, item.behavior, item.placeholder,
-                        item.description, item.layout, item.validations, type);
-                } else if (item.view !== undefined && item.view.value !== undefined && item.view.value === 'area') {
-                    type = TextFieldView.TEXTAREA;
-                } else if (item.view !== undefined && item.view.value !== undefined && item.view.value === 'htmlEditor') {
-                    type = TextFieldView.HTMLTEXTAREA;
-                    return new TextAreaField(item.stringId, item.name, item.value as string, item.behavior, item.placeholder,
-                        item.description, item.layout, item.validations, type);
-                }
-                if (item.component !== undefined && item.component.name !== undefined && (item.component.name === 'editor' ||
-                    item.component.name === TextFieldView.HTMLTEXTAREA || item.component.name === TextFieldView.RICHTEXTAREA)) {
-                    return new TextAreaField(item.stringId, item.name, this.resolveTextValue(item, item.value),
-                        item.behavior, item.placeholder, item.description, item.layout, item.validations, type, item.component,
-                        item.parentTaskId);
+                if (item.component !== undefined && item.component.name !== undefined && (item.component.name === 'textarea' ||
+                    item.component.name === 'richtextarea' || item.component.name === 'htmltextarea' ||
+                    item.component.name === 'editor' || item.component.name === 'htmlEditor' || item.component.name === 'area') ) {
```

Comment:
this if is quite monstrous...

the component name can be safely accessed with `item.component?.name`

the 6 options that result in a TextAreaField should be held in some array or other collection and the condition should just check if the `item.component?.name` is in the collection or not.

Reply:
done